Amazon.QLDB.Driver.Tests/AsyncQldbDriverTests.cs

```diff
@@ -205,45 +164,52 @@ public async Task TestAsyncGetSession_FailedToCreateSession_ThrowTheOriginalExce
             var exception = new AmazonServiceException("test");
             mockClient.QueueResponse(exception);
 
-            var driver = new AsyncQldbDriver("ledgerName", mockClient, 1, NullLogger.Instance);
+            var driver = new AsyncQldbDriver(TestLedger, mockClient, 1, NullLogger.Instance);
 
             try
             {
                 await driver.GetSession();
+                
+                Assert.Fail("driver.GetSession() should have thrown retriable exception");
             }
             catch (RetriableException re)
             {
                 Assert.IsNotNull(re.InnerException);
-                Assert.IsTrue(re.InnerException == exception);
-
-                return;
+                Assert.AreEqual(exception, re.InnerException);
             }
-
-            Assert.Fail("driver.GetSession() should have thrown retriable exception");
         }
 
         [TestMethod]
         public async Task TestAsyncExecuteWithActionLambdaCanInvokeSuccessfully()
         {
-            var driver = new AsyncQldbDriver("ledgerName", mockClient, 4, NullLogger.Instance);
-            await driver.Execute(txn => txn.Execute("testStatement"));
+            try
+            {
+                await testDriver.Execute(async txn => await txn.Execute("testStatement"));
+            }
+            catch (Exception)
+            {
+                Assert.Fail("driver.Execute() should not have thrown exception");
+            }
         }
 
         [TestMethod]
         public async Task TestAsyncExecuteWithActionAndRetryPolicyCanInvokeSuccessfully()
         {
-            var driver = new AsyncQldbDriver("ledgerName", mockClient, 4, NullLogger.Instance);
-
-            await driver.Execute(txn => txn.Execute("testStatement"),
-                Amazon.QLDB.Driver.RetryPolicy.Builder().Build());
+            try
+            {
+                await testDriver.Execute(async txn => await txn.Execute("testStatement"),
```

Comment:
What's missing here is the validation that the execute actually called into the transaction, if txn is a mock you can usually validate that a method was called.

Reply:
Addressed.